### PR TITLE
WIP: Release v1.2.0

### PR DESCRIPTION
**Pull request by nim65s**
Hi,

This PR is here to track the work on #731 
Please don't merge it :)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -90,18 +89,30 @@ SET(BOOST_OPTIONAL_COMPONENTS "")
...
+  src/*.cpp
+  )
 
-# Include and add the main directory
-ADD_SUBDIRECTORY(src)
+IF(UNIX)
+  ADD_LIBRARY(${PROJECT_NAME} SHARED ${${PROJECT_NAME}_SOURCES})
+  SET_TARGET_PROPERTIES(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX)
+  TARGET_INCLUDE_DIRECTORIES(${PROJECT_NAME} PUBLIC $<INSTALL_INTERFACE:include>)
+  TARGET_LINK_LIBRARIES(${PROJECT_NAME} pinocchio::pinocchio
```
Comment (nim65s) [inline CMakeLists.txt:107]: @jcarpent : do you mean here ?
This is the case by default : https://cmake.org/cmake/help/latest/command/target_link_libraries.html#libraries-for-both-a-target-and-its-dependents

File: unittest/CMakeLists.txt [hunk 2]
```diff
@@ -12,8 +12,6 @@ FILE(GLOB_RECURSE ${PROJECT_NAME}_FACTORY_TEST
 IF(UNIX)
   ADD_LIBRARY(${PROJECT_NAME}_unittest SHARED ${${PROJECT_NAME}_FACTORY_TEST})
   SET_TARGET_PROPERTIES(${PROJECT_NAME}_unittest PROPERTIES LINKER_LANGUAGE CXX)
-  PKG_CONFIG_USE_DEPENDENCY(${PROJECT_NAME}_unittest eigen3)
-  PKG_CONFIG_USE_DEPENDENCY(${PROJECT_NAME}_unittest pinocchio)
   TARGET_LINK_LIBRARIES(${PROJECT_NAME}_unittest ${PROJECT_NAME} ${Boost_FILESYSTEM_LIBRARY} ${Boost_SYSTEM_LIBRARY} ${Boost_SERIALIZATION_LIBRARY})
```
Comment (jcarpent) [inline unittest/CMakeLists.txt:15]: You can remove the Boost stuff here.

File: CMakeLists.txt (hunk 1 above)
Comment (jcarpent) [reply CMakeLists.txt:107]: What I meant is that some stuff should be public, some others not.

File: unittest/CMakeLists.txt (hunk 2 above)
Comment (nim65s) [reply unittest/CMakeLists.txt]: done

**Comment by nim65s** [comment]
I'll make a release of example-robot-data first, to get rid the lasts `PKG_CONFIG_USE_DEPENDENCY`

**Comment by cmastalli** [comment]
> I'll make a release of example-robot-data first, to get rid the lasts `PKG_CONFIG_USE_DEPENDENCY`

Please don't merge the last commits regarding the IRIS stuff. If you want to wait a bit, I will make the force push first

**Comment by nim65s** [comment]
@jcarpent : I have checked every test that I had to run on the current state of this PR, and this takes a lot of  time on my side, plus involves waiting several hours, so I usually run this at night.

I suggest to keep your last suggestion for another time, eg. by discussing that in a separate issue, to avoid additional delays everywhere.

Sorry for my laziness x)

**Comment by nim65s** [comment]
https://gepgitlab.laas.fr/loco-3d/crocoddyl/pipelines/8941/builds is also OK. If Travis pass, I'll make the release, without the travis commit, and then I'll cherry-pick the travis commit on devel.

**Comment by cmastalli** [comment]
@nim65s I see the branch `release/v1.2.0`. Do we still need it? Could I remove it?

**Comment by nim65s** [comment]
I removed it 2min before your comment :)

**Pull request closed without merging**